crates/gossip-service/src/cache.rs

```diff
@@ -0,0 +1,145 @@
+use crate::types::{GossipError, GossipResult};
+use irys_types::{BlockHash, ChunkPathHash, GossipData, H256};
+use std::net::SocketAddr;
+use std::ops::DerefMut;
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
+    time::{Duration, Instant},
+};
+
+/// Tracks which peers have seen what data to avoid sending duplicates
+#[derive(Debug, Default)]
+pub struct GossipCache {
+    /// Maps data identifiers to a map of peer IPs and when they last saw the data
+    chunks: Arc<RwLock<HashMap<ChunkPathHash, HashMap<SocketAddr, Instant>>>>,
+    transactions: Arc<RwLock<HashMap<H256, HashMap<SocketAddr, Instant>>>>,
```

Comment:
```suggestion
    transactions: Arc<RwLock<HashMap<IrysTransactionId, HashMap<SocketAddr, Instant>>>>,
```
We have a type alias to make txids a little more distinct in the code.